pydantic/mypy.py

```diff
@@ -75,7 +75,7 @@
 CONFIGFILE_KEY = 'pydantic-mypy'
 METADATA_KEY = 'pydantic-mypy-metadata'
 BASEMODEL_FULLNAME = 'pydantic.main.BaseModel'
-BASESETTINGS_FULLNAME = 'pydantic.env_settings.BaseSettings'
+MODEL_METACLASS_FULLNAME = 'pydantic.main.ModelMetaclass'
```

Comment:
This seems to be an issue of the merge conflict resolution. The PR should only add `MODEL_METACLASS_FULLNAME` not replace `BASESETTINGS_FULLNAME`.

I've opened #5111 with a manual backport. Feel free to use that or close it if you'd like to change it here.